wallet-connect/src/client.rs

```diff
@@ -274,29 +273,27 @@ impl Middleware for WCMiddleware<Provider<Client>> {
             .request("eth_signTransaction", vec![tx_obj])
             .await
             .map_err(|e| WCError::ClientError(ClientError::Eyre(eyre!(e))))?;
+
         let tx_rlp = rlp::Rlp::new(tx_bytes.as_ref());
-        // TODO: check that the decoded request matches the typed transaction content here?
-        let signature = match tx {
-            TypedTransaction::Eip1559(_) => {
-                let decoded_request = Eip1559TransactionRequest::decode_signed_rlp(&tx_rlp);
-                decoded_request
-                    .map(|x| x.1)
-                    .map_err(|e| WCError::ClientError(ClientError::Eyre(eyre!(e))))
-            }
-            TypedTransaction::Eip2930(_) => {
-                let decoded_request = Eip2930TransactionRequest::decode_signed_rlp(&tx_rlp);
-                decoded_request
-                    .map(|x| x.1)
-                    .map_err(|e| WCError::ClientError(ClientError::Eyre(eyre!(e))))
-            }
-            TypedTransaction::Legacy(_) => {
-                let decoded_request = TransactionRequest::decode_signed_rlp(&tx_rlp);
-                decoded_request
-                    .map(|x| x.1)
-                    .map_err(|e| WCError::ClientError(ClientError::Eyre(eyre!(e))))
-            }
-        }?;
 
-        Ok(signature)
+        // get first byte of tx_bytes
+        let first_byte = tx_bytes.to_vec()[0];
```

Comment:
should check that non-empty:
```suggestion
        let first_byte = match tx_rlp.data() {
            Ok(d) if !d.is_empty() => Ok(d[0]),
            _ => Err(WCError::ClientError(ClientError::Eyre(eyre!(
                "empty response data"
            )))),
        }?;

Reply:
ok, i'll modify 
data() is not the first byte, test wallet didn't work
as_raw() is the raw bytes for rlp

should pick raw rlp bytes , which is the same with tx_bytes